Assignment files for week-3

soumya213 opened the pull request:
I have attached all the 4 assignments for week-3

soumya213 committed:
Add files via upload

I have attached all the 4 assignments for week-3

RajeshSivanesan commented on Week-3/assignment-1.js:
```diff
@@ -0,0 +1,45 @@
+
+// https://github.com/soumya213/p4-rajesh-soumya213/pull/3
+
+// Why add function called for the 4th function
```
@soumya213 this we discussed in the 1:1 session. Can you remove this comment. And please change the commit message, its always "Add files via upload". If you uploading each file, please give commit messages based on week and assignment

soumya213 replied:
@RajeshSivanesan removed the comments from `assignment-1.js` and changed the PR title to `Assignment files for week-3`

RajeshSivanesan commented on Week-3/assignment-2.js:
```diff
@@ -0,0 +1,71 @@
+    console.log(`${this.name} have completed assignment ${this.value} ${additionalInformation}.`);
+}
+
+soumya={
```
@soumya213 prefer using let or const, this is global variable. If type is not mentioned it goes to the global under window.

soumya213 replied:
@RajeshSivanesan Added `const` keyword along with below changes: 

Changes made: 

Added `const` keyword to: 
```
const a=10 //Global variable 
const b=10 //Global variable 
```

Added the below if-else clause to:

`function pesto(additionalInformation)` and `function pestoApplyfunction(additionalInformation)` to remove `undefined` from when `additionalInformation` value is not provided

```
if (additionalInformation) {
        console.log(`${this.name} have completed assignment ${this.value} ${additionalInformation}.`);
    }
    else{
    console.log(`${this.name} have completed assignment ${this.value}.`);
    }
```
Added `let` keyword to below objects: 

```
let soumya={
    name:'Soumya',
    value:3
}

let vijay={
    name:'Vijay',
    value:2
}
```

RajeshSivanesan commented on Week-3/assignment-2.js:
```diff
@@ -0,0 +1,71 @@
+
+// Bind
+function multiply(a,b) {
+
```
@soumya213 multiply is a simple function which accepts a and b. If we pass different values to a & b it will give its corresponding value. Do you think bind is needed here. 

Can you define a object and have some methods, and borrow that method for a another object by binding.

soumya213 replied:
@RajeshSivanesan I have updated the Bind function methods

```
// Bind

let name1={
    name:'Soumya',
    title:'Dhar',
    
    works: function (workplace) {
        console.log(this.name + ' ' + this.title + ' is working ' + workplace);
    }
}

let name2={
    name:'Sthiti',
    title:'Nath',
}

let name3={
    name:'Abhsekh',
    title:'Panigrahi',
}

workPlace= name1.works.bind(name2,'tcs')
workPlace()

workPlace= name1.works.bind(name3,'pwc')
workPlace()

/*
 bind() method doesn’t immediately execute the function. It just returns a new version of the function whose this sets to thisArg argument. 
*/
```

RajeshSivanesan commented on Week-3/assignment-3.js:
```diff
@@ -0,0 +1,27 @@
+
+// Here the increment function is called thrice, so the count will be incremnted by 3 and finally the count value will be 3.
+
+// But the console.log value will be 0 because variable `message` is a not incremented and assigned a value of 0 at the start.
```
@soumya213 message is not incremented (statement seems wrong). You can mention that the message is initialised only once during parent function call and its not getting reset, hence the value is 0

soumya213 replied:
@RajeshSivanesan Modified the explanation of the message to: 

/*
The increment function is called thrice, so the count will be incremented by 3 and finally the count value will be 3.
But the console.log value for message is initialized only once during parent function call and its not getting reset, hence the value is 0.
*/

RajeshSivanesan reviewed (commented):
@soumya213 Looks good. Added some comments. Please check.

soumya213 committed:
Update assignment-1.html

Changed title from `<title>Assignment-3.1</title>` to `<title>Assignment-3</title>`

soumya213 committed:
Update assignment-1.js

Removed the below comments as this was discussed in 1:1

// https://github.com/soumya213/p4-rajesh-soumya213/pull/3

// Why add function called for the 4th function

soumya213 committed:
Update assignment-2.js

Changes made: 

Added `const` keyword to: 
```
const a=10 //Global variable 
const b=10 //Global variable 
```

Added the below if-else clause to:

`function pesto(additionalInformation)` and `function pestoApplyfunction(additionalInformation)` to remove `undefined` from when `additionalInformation` value is not provided

```
if (additionalInformation) {
        console.log(`${this.name} have completed assignment ${this.value} ${additionalInformation}.`);
    }
    else{
    console.log(`${this.name} have completed assignment ${this.value}.`);
    }
```
Added `let` keyword to below objects: 

```
let soumya={
    name:'Soumya',
    value:3
}

let vijay={
    name:'Vijay',
    value:2
}
```

soumya213 committed:
Update assignment-3.js

Modified the explanation of the message to: 

/*
The increment function is called thrice, so the count will be incremnted by 3 and finally the count value will be 3.
But the console.log value for message is initialised only once during parent function call and its not getting reset, hence the value is 0.
*/

soumya213 committed:
Update assignment-3.js

soumya213 committed:
Updated Bind method

```
// Bind

let name1={
    name:'Soumya',
    title:'Dhar',
    
    works: function (workplace) {
        console.log(this.name + ' ' + this.title + ' is working ' + workplace);
    }
}

let name2={
    name:'Sthiti',
    title:'Nath',
}

let name3={
    name:'Abhsekh',
    title:'Panigrahi',
}

workPlace= name1.works.bind(name2,'tcs')
workPlace()

workPlace= name1.works.bind(name3,'pwc')
workPlace()

/*
 bind() method doesn’t immediately execute the function. It just returns a new version of the function whose this sets to thisArg argument. 
*/
```
